frontend/lib/resolution/Resolver.cpp

```diff
@@ -3492,6 +3492,10 @@ void Resolver::exit(const Dot* dot) {
   }
 
   if (dot->field() == USTR("type")) {
+    if (receiver.type().isType()) {
+      CHPL_REPORT(context, DotTypeOnType, dot);
```

Comment:
Probably need to set the type to `CHPL_TYPE_ERROR`  instead of just calling `CHPL_REPORT` to make sure we propagate `ErrorneousType`s